code/modules/mob/living/simple_animal/hostile/ordeal/gold.dm

```diff
@@ -723,9 +725,529 @@
+
+/mob/living/simple_animal/hostile/ordeal/snake_corrosion/proc/SpawnPoison()
+	var/turf/target_turf = get_turf(src)
+	for(var/turf/T in view(poison_range, target_turf))
+		if(prob(30))
+			continue
+		new /obj/effect/temp_visual/mustardgas(T)
+		for(var/mob/living/H in T)
+			if(faction_check_mob(H))
+				continue
+			H.apply_damage(poison_damage, BLACK_DAMAGE, null, H.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
+			H.apply_venom(2)
+
+/mob/living/simple_animal/hostile/ordeal/snake_corrosion/apply_damage(damage, damagetype, def_zone, blocked, forced, spread_damage, wound_bonus, bare_wound_bonus, sharpness, white_healable)
+	. = ..()
+	if(poison_releasing)
+		return
+	if(health <= (maxHealth * 0.5))
+		poison_releasing = TRUE
+		visible_message(span_warning("[src]'s body breaks, releasing toxic fumes!"), span_boldwarning("Your poison gland breaks!"))
+		playsound(src, 'sound/abnormalities/doomsdaycalendar/Limbus_Dead_Generic.ogg', 40, 0, FALSE)
+		rapid += 1
+
+/mob/living/simple_animal/hostile/ordeal/snake_corrosion/Move()
+	if(!can_act)
+		return FALSE
+	..()
+
+/mob/living/simple_animal/hostile/ordeal/snake_corrosion/AttackingTarget()
+	if(!can_act)
+		return FALSE
+	..()
+	if(isliving(target))
+		var/mob/living/H = target
+		H.apply_venom(applied_venom)
+
+/mob/living/simple_animal/hostile/ordeal/snake_corrosion/OpenFire()
+	if(get_dist(src, target) > 4)
+		return
+	. = ..()
+	can_act = FALSE
+	guntimer = addtimer(CALLBACK(src, PROC_REF(StartMoving)), (5), TIMER_STOPPABLE)
+
+/mob/living/simple_animal/hostile/ordeal/snake_corrosion/proc/StartMoving()
+	can_act = TRUE
+	deltimer(guntimer)
+
+/mob/living/simple_animal/hostile/ordeal/snake_corrosion/death(gibbed)
+	. = ..()
+	if(guntimer)
+		deltimer(guntimer)
+	poison_releasing = FALSE
+
+/mob/living/simple_animal/hostile/ordeal/snake_corrosion/bullet_act(obj/projectile/P)
+	if(!P.firer)
+		return ..()
+	if((get_dist(P.firer, src) > 4))
+		new /obj/effect/temp_visual/healing/no_dam(get_turf(src))
+		visible_message(span_userdanger("[src] blocks \the [P]!"))
+		P.Destroy()
+	return ..()
+
+#define STATUS_EFFECT_VENOM /datum/status_effect/stacking/venom
+/datum/status_effect/stacking/venom
+	id = "stacking_venom"
+	status_type = STATUS_EFFECT_UNIQUE
+	alert_type = /atom/movable/screen/alert/status_effect/venom
+	max_stacks = 10
+	tick_interval = 5 SECONDS
+	consumed_on_threshold = FALSE
+
+/atom/movable/screen/alert/status_effect/venom
+	name = "Venom"
+	desc = "Your veins feel like they are on fire!"
+	icon = 'ModularTegustation/Teguicons/status_sprites.dmi'
+	icon_state = "venom"
+
+/datum/status_effect/stacking/venom/can_have_status()
+	return (owner.stat != DEAD || !(owner.status_flags & GODMODE))
+
+/datum/status_effect/stacking/venom/on_apply()
+	. = ..()
+	owner.playsound_local(owner, 'sound/effects/venom_apply.ogg', 50, TRUE)
+	to_chat(owner, "<span class='warning'>You have been envenomed!</span>")
+
+//Deals brute damage, maybe one day we can make it deal toxin instead
```

Comment:
If you make it deal a very small amount of toxin, it can work as a semi-permanent damage type.

Reply:
This wasn't meant to be an issue; one sec